plan/plan.go

```diff
@@ -170,6 +172,11 @@ func (p *Plan) Calculate() *Plan {
 		changes = pol.Apply(changes)
 	}
 
+	// Handle the migration of the TXT records created before the new format (introduced in v0.12.0)
+	if len(p.Missing) > 0 {
+		changes.Create = append(changes.Create, filterRecordsForPlan(p.Missing, p.DomainFilter, append(p.ManagedRecords, endpoint.RecordTypeTXT))...)
+	}
```

Comment:
Though not necessary to do so, it is safe to call `filterRecordsForPlan` with an empty argument for its "records" parameter. It will do no extra work, though you'd wind up calling `append` two more times.

Reply:
Not quite sure I understood your suggestion. Can you please show the code of how you think it can look like?

Reply:
You _could_ write the following:
```go
	changes.Create = append(changes.Create, filterRecordsForPlan(p.Missing, p.DomainFilter, append(p.ManagedRecords, endpoint.RecordTypeTXT))...)
```
That is, drop the guard condition on `p.Missing` being nonempty.

The way you wrote it is a little more efficient. You wind up testing the length of `p.Missing` twice when it's nonempty, but save on two calls to `append` when it is empty.

Reply:
Ah yes, true, the slice of missing endpoint can be passed without the guard check, thanks for the example!
Yeah, I think it's better to avoid the unnecessary CPU cycles in the empty case.